src/window.rs

```diff
@@ -168,7 +171,44 @@ impl WindowWrapper {
             previous_dpis,
             transparency: 1.0,
             fullscreen: false,
+            cached_width: 0,
+            cached_height: 0,
+            cached_position: (0, 0)
+        }
+    }
+
+    pub fn toggle_fullscreen(&mut self) {
+        unsafe {
+            let raw_handle = self.window.raw();
+            let display_index = sdl2::sys::SDL_GetWindowDisplayIndex(raw_handle);
+
+            if let Ok(rect) = self.window.subsystem().display_bounds(display_index) {    
+                if self.fullscreen {
+                    // Set window back to resizable
+                    sdl2::sys::SDL_SetWindowResizable(raw_handle, sdl2::sys::SDL_bool::SDL_TRUE);
+
+                    // Use cached size and position
+                    self.window.set_size(self.cached_width as u32, self.cached_height as u32).unwrap();
```

Comment:
I like this and I think it is a marked improvement over the current experience. I do think it should be integrated with the setting system that already exists though. I don't feel comfortable merging until then.

Reply:
By integration do you mean an option to either have the old fullscreen mode or have the toggleable mode?

Reply:
Its already toggleable, but has bugs. It can be enabled or disabled by setting g:neovide_fullscreen to v:true or v:false. I'm suggesting integrating the bug fixes with that existing system.

Reply:
Ah I see, if I understand you correctly, I'll just try to integrate the functionality with the existing setting so that users can just keybind a true/false toggle in their .vimrc